src/pdiagmat.jl

```diff
@@ -2,11 +2,14 @@
 Positive definite diagonal matrix.
 """
 struct PDiagMat{T<:Real,V<:AbstractVector{T}} <: AbstractPDMat{T}
-    dim::Int
     diag::V
 end
 
-PDiagMat(v::AbstractVector{<:Real}) = PDiagMat{eltype(v),typeof(v)}(length(v), v)
+function Base.getproperty(a::PDiagMat, s::Symbol)
+    s === :dim && return length(getfield(a, :diag))
```

Comment:
I think I'd prefer
```julia
    if s === :dim
        return length(getfield(a, :diag))
    else
        ...
    end
```
It's mostly a matter of personal preference but it also had happened to me that the `.... && ...` form masked a code coverage loss in cases where the branch was not tested at all.

Reply:
Can I suggest adding a `.JuliaFormatter.toml` to enforce style suggestions, so that we don't have to bother discussing such minutiae on PRs?

Reply:
As I said, I think it would be better for coverage analysis. I'd prefer a JuliaFormatter setting (even though I'm not sure if it would help here) but unfortuantely I expect that one has to deal with such questions in JuliaStats PRs for the foreseeable future. Discussions around JuliaFormatter haven't lead to any consensus yet, seems people have strong opinions about formatting code 😅

Reply:
My strongest opinion on formatting is that we should have `.JuliaFormatter.toml`s almost irrespective of what those settings are, to (a) enforce consistency and (b) avoid these discussions and debates. Bike shedding should be called out for the waste of time it is.
Anyway, I did convert these branches.

Maybe I missed it, but I unfortunately didn't see a setting for this
https://domluna.github.io/JuliaFormatter.jl/dev/

Reply:
https://github.com/domluna/JuliaFormatter.jl/issues/759